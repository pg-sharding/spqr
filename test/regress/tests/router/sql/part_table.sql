
-- Check partitioned tables.
\c spqr-console

-- SETUP
CREATE DISTRIBUTION ds1 COLUMN TYPES varchar hash;

CREATE KEY RANGE krid1 FROM 0 ROUTE TO sh1 FOR DISTRIBUTION ds1;
CREATE KEY RANGE krid2 FROM 2147483648 ROUTE TO sh2 FOR DISTRIBUTION ds1;

-- the set of all unsigned 32-bit integers (0 to 4294967295)
ALTER DISTRIBUTION ds1 ATTACH RELATION xxhash_part DISTRIBUTION KEY i HASH FUNCTION MURMUR;


\c regress
CREATE TABLE xxhash_part (i uuid, d date) partition by range (d);

create table xxhash_part_1 partition of xxhash_part for values from ('2024-12-01') to ('2024-12-31');
create table xxhash_part_2 partition of xxhash_part for values from ('2024-11-01') to ('2024-11-30');
create table xxhash_part_3 partition of xxhash_part for values from ('2024-10-01') to ('2024-10-31');

insert into xxhash_part (i,d ) values ('b37da10c-25a0-4473-96fd-c3871bd81d00', '2024-12-23');
insert into xxhash_part (i,d ) values ('b1569ed9-4029-495f-8d7c-02b216528c0a', '2024-12-21');
insert into xxhash_part (i,d ) values ('0583484b-e407-41e3-9c75-66c1994944c8', '2024-12-20');
insert into xxhash_part (i,d ) values ('b14f8871-ba67-4244-8fa6-45edabaf6206', '2024-12-22');

insert into xxhash_part (i,d ) values ('7906fd75-60ae-4e1d-99c5-da19d4b63515', '2024-11-23');
insert into xxhash_part (i,d ) values ('99d75dea-c62c-4bb3-9d6e-eda36432bda0', '2024-11-21');
insert into xxhash_part (i,d ) values ('0bd49f01-b01f-4c18-bd44-d75746a7c71c', '2024-11-20');
insert into xxhash_part (i,d ) values ('3dcd025f-b016-4d1c-aaa8-4f8462f9aa8e', '2024-11-22');

insert into xxhash_part (i,d ) values ('93043bc0-b6d3-4bb6-b1b4-869d2b4a4967', '2024-10-23');
insert into xxhash_part (i,d ) values ('f6a01c78-9ce8-4cc3-a78f-32bebc4ccb4a', '2024-10-21');
insert into xxhash_part (i,d ) values ('b3e31fad-80fa-48df-bcc0-51ffd1594df3', '2024-10-20');
insert into xxhash_part (i,d ) values ('d15707df-f256-49a5-a693-3feda5afe678', '2024-10-22');


SELECT * FROM xxhash_part ORDER BY d /* __spqr__execute_on: sh1 */;
SELECT * FROM xxhash_part ORDER BY d /* __spqr__execute_on: sh2 */;

--TEARDOWN
DROP TABLE xxhash_part;
\c spqr-console
DROP DISTRIBUTION ALL CASCADE;
DROP KEY RANGE ALL;
