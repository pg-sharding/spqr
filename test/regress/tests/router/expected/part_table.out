-- Check partitioned tables.
\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

-- SETUP
CREATE DISTRIBUTION ds1 COLUMN TYPES varchar hash;
    add distribution    
------------------------
 distribution id -> ds1
(1 row)

CREATE KEY RANGE krid1 FROM 0 ROUTE TO sh1 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 0
(1 row)

CREATE KEY RANGE krid2 FROM 2147483648 ROUTE TO sh2 FOR DISTRIBUTION ds1;
    add key range    
---------------------
 bound -> 2147483648
(1 row)

-- the set of all unsigned 32-bit integers (0 to 4294967295)
ALTER DISTRIBUTION ds1 ATTACH RELATION xxhash_part DISTRIBUTION KEY i HASH FUNCTION MURMUR;
          attach table          
--------------------------------
 relation name   -> xxhash_part
 distribution id -> ds1
(2 rows)

\c regress
CREATE TABLE xxhash_part (i uuid, d date) partition by range (d);
NOTICE: send query to shard(s) : sh1,sh2
create table xxhash_part_1 partition of xxhash_part for values from ('2024-12-01') to ('2024-12-31');
NOTICE: send query to shard(s) : sh1,sh2
create table xxhash_part_2 partition of xxhash_part for values from ('2024-11-01') to ('2024-11-30');
NOTICE: send query to shard(s) : sh1,sh2
create table xxhash_part_3 partition of xxhash_part for values from ('2024-10-01') to ('2024-10-31');
NOTICE: send query to shard(s) : sh1,sh2
insert into xxhash_part (i,d ) values ('b37da10c-25a0-4473-96fd-c3871bd81d00', '2024-12-23');
NOTICE: send query to shard(s) : sh2
insert into xxhash_part (i,d ) values ('b1569ed9-4029-495f-8d7c-02b216528c0a', '2024-12-21');
NOTICE: send query to shard(s) : sh1
insert into xxhash_part (i,d ) values ('0583484b-e407-41e3-9c75-66c1994944c8', '2024-12-20');
NOTICE: send query to shard(s) : sh2
insert into xxhash_part (i,d ) values ('b14f8871-ba67-4244-8fa6-45edabaf6206', '2024-12-22');
NOTICE: send query to shard(s) : sh1
insert into xxhash_part (i,d ) values ('7906fd75-60ae-4e1d-99c5-da19d4b63515', '2024-11-23');
NOTICE: send query to shard(s) : sh2
insert into xxhash_part (i,d ) values ('99d75dea-c62c-4bb3-9d6e-eda36432bda0', '2024-11-21');
NOTICE: send query to shard(s) : sh2
insert into xxhash_part (i,d ) values ('0bd49f01-b01f-4c18-bd44-d75746a7c71c', '2024-11-20');
NOTICE: send query to shard(s) : sh2
insert into xxhash_part (i,d ) values ('3dcd025f-b016-4d1c-aaa8-4f8462f9aa8e', '2024-11-22');
NOTICE: send query to shard(s) : sh1
insert into xxhash_part (i,d ) values ('93043bc0-b6d3-4bb6-b1b4-869d2b4a4967', '2024-10-23');
NOTICE: send query to shard(s) : sh1
insert into xxhash_part (i,d ) values ('f6a01c78-9ce8-4cc3-a78f-32bebc4ccb4a', '2024-10-21');
NOTICE: send query to shard(s) : sh1
insert into xxhash_part (i,d ) values ('b3e31fad-80fa-48df-bcc0-51ffd1594df3', '2024-10-20');
NOTICE: send query to shard(s) : sh2
insert into xxhash_part (i,d ) values ('d15707df-f256-49a5-a693-3feda5afe678', '2024-10-22');
NOTICE: send query to shard(s) : sh2
SELECT * FROM xxhash_part ORDER BY d /* __spqr__execute_on: sh1 */;
NOTICE: send query to shard(s) : sh1
                  i                   |     d      
--------------------------------------+------------
 f6a01c78-9ce8-4cc3-a78f-32bebc4ccb4a | 2024-10-21
 93043bc0-b6d3-4bb6-b1b4-869d2b4a4967 | 2024-10-23
 3dcd025f-b016-4d1c-aaa8-4f8462f9aa8e | 2024-11-22
 b1569ed9-4029-495f-8d7c-02b216528c0a | 2024-12-21
 b14f8871-ba67-4244-8fa6-45edabaf6206 | 2024-12-22
(5 rows)

SELECT * FROM xxhash_part ORDER BY d /* __spqr__execute_on: sh2 */;
NOTICE: send query to shard(s) : sh2
                  i                   |     d      
--------------------------------------+------------
 b3e31fad-80fa-48df-bcc0-51ffd1594df3 | 2024-10-20
 d15707df-f256-49a5-a693-3feda5afe678 | 2024-10-22
 0bd49f01-b01f-4c18-bd44-d75746a7c71c | 2024-11-20
 99d75dea-c62c-4bb3-9d6e-eda36432bda0 | 2024-11-21
 7906fd75-60ae-4e1d-99c5-da19d4b63515 | 2024-11-23
 0583484b-e407-41e3-9c75-66c1994944c8 | 2024-12-20
 b37da10c-25a0-4473-96fd-c3871bd81d00 | 2024-12-23
(7 rows)

--TEARDOWN
DROP TABLE xxhash_part;
NOTICE: send query to shard(s) : sh1,sh2
\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

DROP DISTRIBUTION ALL CASCADE;
   drop distribution    
------------------------
 distribution id -> ds1
(1 row)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

